src/System.Runtime.Extensions/src/System/Environment.Windows.cs

```diff
@@ -0,0 +1,290 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        private static readonly Lazy<bool> s_isAppX = new Lazy<bool>(() =>
+        {
+            // TODO: Determine the right way to get at this information from outside of System.Private.Corelib
```

Comment:
A better alternative may be to do different builds of `System.Runtime.Extensions.dll` for appx vs. console. It is what we are doing for all other custom appx-specific implementations in corefx today.